logrotate/src/LogRotateHelper.ipp

```diff
@@ -267,11 +267,13 @@ std::string LogRotateHelper::changeLogFile(const std::string& directory, const s
     std::string file_name = new_name;
     if (file_name.empty()) {
         std::cout << "no filename" << std::endl;
-       file_name = createLogFileName(log_prefix_backup_);
+        file_name = log_prefix_backup_;
+    } else {
+        log_prefix_backup_ = file_name;
```

Comment:
After line 281 we know the file name is good. After this we can change the default backup